zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -19,6 +25,240 @@ use {
+                id_note INTEGER PRIMARY KEY,
+                tx INTEGER NOT NULL,
+                output_index INTEGER NOT NULL,
+                from_account INTEGER NOT NULL,
+                address TEXT NOT NULL,
+                value INTEGER NOT NULL,
+                memo BLOB,
+                FOREIGN KEY (tx) REFERENCES transactions(id_tx),
+                FOREIGN KEY (from_account) REFERENCES accounts(account),
+                CONSTRAINT tx_output UNIQUE (tx, output_index)
+            );",
+        )?;
+        Ok(())
+    }
+
+    fn down(&self, _transaction: &Transaction) -> Result<(), SqliteClientError> {
+        // We should never down-migrate the first migration, as that can irreversibly
+        // destroy data.
+        panic!("Cannot revert the initial migration.");
+    }
+}
+
+struct WalletMigration1;
+
+migration!(
+    WalletMigration1,
+    "a2e0ed2e-8852-475e-b0a4-f154b15b9dbe",
+    ["bc4f5e57-d600-4b6c-990f-b3538f0bfce1"],
+    "Add support for receiving transparent UTXOs."
+);
+
+impl RusqliteMigration for WalletMigration1 {
+    type Error = SqliteClientError;
+
+    fn up(&self, transaction: &Transaction) -> Result<(), SqliteClientError> {
+        transaction.execute_batch(
+            "CREATE TABLE IF NOT EXISTS utxos (
+                id_utxo INTEGER PRIMARY KEY,
+                address TEXT NOT NULL,
+                prevout_txid BLOB NOT NULL,
+                prevout_idx INTEGER NOT NULL,
+                script BLOB NOT NULL,
+                value_zat INTEGER NOT NULL,
+                height INTEGER NOT NULL,
+                spent_in_tx INTEGER,
+                FOREIGN KEY (spent_in_tx) REFERENCES transactions(id_tx),
+                CONSTRAINT tx_outpoint UNIQUE (prevout_txid, prevout_idx)
+            );",
+        )?;
+        Ok(())
+    }
+
+    fn down(&self, transaction: &Transaction) -> Result<(), SqliteClientError> {
+        transaction.execute_batch("DROP TABLE utxos;")?;
+        Ok(())
+    }
+}
+
+struct WalletMigration2<P: consensus::Parameters> {
+    params: P,
+    seed: Vec<u8>,
+}
+
+impl<P: consensus::Parameters> Migration for WalletMigration2<P> {
+    fn id(&self) -> Uuid {
+        ::uuid::Uuid::parse_str("be57ef3b-388e-42ea-97e2-678dafcf9754").unwrap()
+    }
+
+    fn dependencies(&self) -> HashSet<Uuid> {
+        ["a2e0ed2e-8852-475e-b0a4-f154b15b9dbe"]
+            .iter()
+            .map(|uuidstr| ::uuid::Uuid::parse_str(uuidstr).unwrap())
+            .collect()
+    }
+
+    fn description(&self) -> &'static str {
+        "Add support for unified full viewing keys"
+    }
+}
+
+impl<P: consensus::Parameters> RusqliteMigration for WalletMigration2<P> {
+    type Error = SqliteClientError;
+
+    fn up(&self, transaction: &Transaction) -> Result<(), SqliteClientError> {
+        //
+        // Update the accounts table to store ufvks rather than extfvks
+        //
+
+        transaction.execute_batch(
+            "PRAGMA foreign_keys = OFF;
```

Comment:
Would it make sense to move the pragmas into their own transaction?

I'm just thinking of making it visually easier for humans to balance the foreign keys being turned on/off here and at the very end.  (The one at the end is easy to miss since its tucked inside a bunch of other SQL).

Reply:
Everything in a migration runs in a single transaction, with multiple statements. I'd be fine with splitting the text here, but it's purely a textual change, so I didn't see much of a reason for it.

Reply:
Yes this is correct from the machine perspective.

It is a suggestion for humans reading the code, rather than that machine.  

If we ever have to modify this code, I want the human reading it to be able to balance the `"PRAGMA foreign_keys = OFF;` at the beginning with another one at the end.  Right now, these statements hide and don't stand out so a human reader could miss it.

Reply:
This is non-blocking, but a small change here improves readability.